src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -209,7 +210,7 @@ def format_annotation(annotation: Any, config: Config) -> str:  # noqa: C901, PL
     fully_qualified: bool = getattr(config, "typehints_fully_qualified", False)
     prefix = "" if fully_qualified or full_name == class_name else "~"
     role = "data" if module == "typing" and class_name in _PYDATA_ANNOTATIONS else "class"
-    args_format = "\\[{}]"
+    args_format = "\\ \\[{}]"
```

Comment:
Maybe it would be better to use `r` here and in lots of other places: `r"\ \[{}]"` would have half as many slashes...